raster/src/main/scala/geotrellis/raster/io/geotiff/tags/GeoKeyReader.scala

```diff
@@ -34,10 +34,11 @@ object GeoKeyReader {
   ): GeoKeyDirectory = {
 
     def readGeoKeyEntry(keyMetadata: GeoKeyMetadata,
-      geoKeyDirectory: GeoKeyDirectory)  = keyMetadata.tiffTagLocation match {
-      case 0 => readShort(keyMetadata, geoKeyDirectory)
-      case DoublesTag => readDoubles(keyMetadata, geoKeyDirectory)
-      case AsciisTag => readAsciis(keyMetadata, geoKeyDirectory)
+      geoKeyDirectory: GeoKeyDirectory): Option[GeoKeyDirectory] = keyMetadata.tiffTagLocation match {
+      case 0 => Some(readShort(keyMetadata, geoKeyDirectory))
+      case DoublesTag => Some(readDoubles(keyMetadata, geoKeyDirectory))
+      case AsciisTag => Some(readAsciis(keyMetadata, geoKeyDirectory))
+      case _ => None
```

Comment:
Can you put a comment on the `readGeoKeyEntry` method that basically explains the same reasoning as in the PR description? This will save some future version of ourselves time otherwise spent on thinking through why this would be returning an option.

Reply:
Good point, addressed in https://github.com/locationtech/geotrellis/pull/2531/commits/788a03464d05420ebf6f97da9b5b170d223e2072